packages/fxa-auth-server/lib/routes/attached-clients.js

```diff
@@ -0,0 +1,254 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+const isA = require('joi');
+const validators = require('./validators');
+const error = require('../error');
+
+const HEX_STRING = validators.HEX_STRING;
+const DEVICES_SCHEMA = require('../devices').schema;
+
+module.exports = (log, db, oauthdb, devices, clientUtils) => {
+
+  return [
+    {
+      method: 'GET',
+      path: '/account/attached_clients',
+      options: {
+        auth: {
+          strategy : 'sessionToken'
+        },
+        response: {
+          schema: isA.array().items(isA.object({
+            clientId: isA.string().regex(HEX_STRING).allow(null).required(),
+            deviceId: DEVICES_SCHEMA.id.allow(null).required(),
+            sessionTokenId: isA.string().regex(HEX_STRING).allow(null).required(),
+            refreshTokenId: isA.string().regex(HEX_STRING).allow(null).required(),
+            isCurrentSession: isA.boolean().required(),
+            deviceType: DEVICES_SCHEMA.type.allow(null).required(),
+            name: DEVICES_SCHEMA.nameResponse.allow('').allow(null).required(),
+            createdTime: isA.number().min(0).required().allow(null),
+            createdTimeFormatted: isA.string().optional().allow(''),
+            lastAccessTime: isA.number().min(0).required().allow(null),
+            lastAccessTimeFormatted: isA.string().optional().allow(''),
+            approximateLastAccessTime: isA.number().min(0).optional(),
+            approximateLastAccessTimeFormatted: isA.string().optional().allow(''),
+            scope: isA.array().items(validators.scope).required().allow(null),
+            location: DEVICES_SCHEMA.location,
+            userAgent: isA.string().max(255).required().allow(''),
+            os: isA.string().max(255).allow('').allow(null),
+          }))
+        }
+      },
+      handler: async function (request) {
+        log.begin('Account.attachedClients', request);
+
+        const uid = request.auth.credentials.uid;
+
+        const attachedClients = [];
+
+        const defaultFields = {
+          clientId: null,
+          deviceId: null,
+          sessionTokenId: null,
+          refreshTokenId: null,
+          isCurrentSession: false,
+          deviceType: null,
+          name: null,
+          createdTime: Infinity,
+          lastAccessTime: 0,
+          scope: null,
+          location: null,
+          userAgent: '',
+          os: null,
+          //scopes: null,
```

Comment:
is this left for later?

Reply:
whoops, accidental dupe of a not-commented-out field.